DINA-Web/dina-collections
# Improve scripts - checkout tag branch, rebuild elastic, remote exec cmd

## Checkout tag branch:

To checkout a specific tag on a server

To test this:

You first need to make sure that the demo server is on this branch: anton/add-script-to-rebuild-search-index 

To verify run:
```
yarn admin:print:server-status -s demo
```

Then to checkout a specific tag run:
``` 
yarn admin:git:checkout-tag -s demo -t v0.15.1
```
Then again run:
```
yarn admin:print:server-status -s demo
```
You should see get branch tag  =  HEAD       v0.15.1 

If you want to test this again you need to change the branch back to anton/add-script-to-rebuild-search-index 


## Rebuild elastic
To rebuild elastic locally run:

```
yarn index:specimen-search
```

To rebuild elastic on a server run: 

```
yarn admin:index:specimen-search -s demo -t v0.15.1
```
Since this is not in the docker image it will not work so dont mind this :). 


## Remote exec cmd

Ex:
To get the hostname of the server.
```
yarn admin:exec:cmd -s demo -c 'hostname'
```

To inspect a specific error.
```
yarn admin:exec:cmd -s demo -c 'docker-compose logs --tail=2000 api | grep -a20 -b20 7ac60732-a396-434c-b59c-a864cd6a97d0'
```
